keras/engine/training.py

```diff
@@ -1835,30 +1837,29 @@ def test_function(inputs):
                         for s, x in zip(self.inputs, inputs[:self._num_data])]
                 label = [K.mx.nd.array(x, dtype=s.dtype) \
                          for s, x in zip(self.targets + self.sample_weights, inputs[self._num_data:])]
-                if not self._train_mod.binded:
+                if not self._test_mod.binded:
```

Comment:
I merged too fast. This was intentional
we can actually use the same mod  for training and testing